text/0000-manifest-list-buildpacks.md

```diff
@@ -0,0 +1,133 @@
+
+The number one question I get about [paketo] buildpacks is whether they support arm. It is safe to say that workloads are generally moving to arm processors and end users are already evaluating how to build containers on arm. It is difficult for me as an end user to seriously consider building arm images with paketo buildpacks without putting in some engineering effort into building arm versions of the paketo buildpacks I may need. While the above example is definitely helpful, a lot of the steps can be built into the github actions workflows that create and release buildpacks for general consumption. I believe arm support was the number one requested roadmap item for 2023, and this proposal is an attempt to jump start that process.
+
+
+## Detailed Explanation
+
+Right now binaries are created on/for amd64 and then packaged up into buildpacks using pack. It should be easy to create binaries for arm64 which can then be packaged up into buildpacks using pack. These architecture-specific buildpacks can be pushed to a registry with a platform specific tag as shown in the examples below. 
+
+* `paketobuildpacks/ca-certificates:3.6-amd64`
+* `paketobuildpacks/ca-certificates:3.6-arm64`
+
+The last step would be to create a manifest list image using `docker manifest` commands (or similar) and push it to the registry.
+
+Users would pull the manifest list image such as: `paketobuildpacks/ca-certificates:3.6`
+
+This is what the manifest might look like for `paketobuildpacks/ca-certificates:3.6` (as an example)
+
+```
+# crane manifest paketobuildpacks/ca-certificates:3.6 | jq .
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
+  "manifests": [
+    {
+      "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
+      "size": 12345,
+      "digest": "sha256:abcdefchijklmnodka0204kdkbjladkj02jfjbe2458801ekdbbks024f555kdkd",
+      "platform": {
+        "architecture": "amd64",
+        "os": "linux"
+      }
+    },
+    {
+      "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
+      "size": 12345,
+      "digest": "sha256:3333defchijklmnodka0204kdkbjladkj02jfjbe2458801ekdbbks024f555kdkd",
+      "platform": {
+        "architecture": "arm64",
+        "os": "linux"
+      }
+    }
+  ]
+}
+# 
+```
+
+## Rationale and Alternatives
+
+The upstream pack project is already creating multi-arch images for pack and the lifecycle using `docker buildx`, which creates manifest list images. An alternative approach is to create and publish separate images that are tagged by architecture. This would be the steps I mentioned above without the final manifest list image. I think this is an anti-pattern at this point given that multi-arch images are already a norm. I also suspect end users would end up creating their own manifest list images anyway. I think the biggest driver for this is to avoid end users cloning and rebuilding paketo buildpacks.
+
+## Implementation
```

Comment:
This RFC will need to be coordinated with #287 because that will move the location of the dependency metadata out of buildpack.toml into a dynamically-discovered location.

I think the two RFCs are compatible, but we will need to ensure that changes proposed by the first RFC to be accepted are reflected in the second RFC.

I don't think this is a problem, but I think we should be aware of it.

cc @paketo-buildpacks/steering-committee for keeping track of RFCs that have potential to overlap.

Reply:
Yup we talked about this in the WG of 2023-04-27 we are going to keep an eye on both of these and adapt accordingly.